src/jit/regalloc.cpp

```diff
@@ -664,11 +664,29 @@ void                Compiler::raSetupArgMasks(RegState *regState)
 // by linear scan. (It is not shared for System V AMD64 platform.)
 regNumber     Compiler::raUpdateRegStateForArg(RegState *regState, LclVarDsc *argDsc)
 {
-    regNumber inArgReg = argDsc->lvArgReg;
+    regNumber inArgReg  = argDsc->lvArgReg;
+    regMaskTP inArgMask = genRegMask(inArgReg);
 
-    noway_assert(genRegMask(inArgReg) & (regState->rsIsFloat ? RBM_FLTARG_REGS : RBM_ARG_REGS));
+    if (regState->rsIsFloat)
+    {
+        noway_assert(inArgMask & RBM_FLTARG_REGS);
+    }
+    else //  regState is for the integer registers
+    {
+        // This might be the fixed return buffer register argument (on ARM64)
+        // We check and allow inArgReg to be theFixedRetBuffReg
+        if (hasFixedRetBuffReg() && (inArgReg == theFixedRetBuffReg()))
+        {
+            // We should have a TYP_BYREF arg not a TYP_STRUCT arg
+            noway_assert(argDsc->lvType != TYP_STRUCT);
```

Comment:
On Arm64 it looks like RBM_ARG_REGS more of a dynamic thing rather than a constant.  We will have a need to know the actual RBM_ARG_REGS that includes x8 if the method has retBufArg at more than one place.  I am wondering would it be a good idea to add an abstraction to get Arg reg set

e.g.
regMaskTP  Compiler::compArgRegMask()
{
    regMaskTP result = RBM_ARG_REGS;

```
#ifdef _TARGET_AMD64_
   if (info.retBuffArg != BAD_VAR_NUM)
   {
         result |= REG_ARG_RET_BUFF
   }
#endif

   return result;
}
```

Now the noway assert will simply become

```
regMaskTP argRegMask = compArgRegMask()
noway_assert(genRegMask(inArgReg) & (regState->rsIsFloat ? RBM_FLTARG_REGS : argRegMask));
```